tests/target/chains_long_items_visual.rs

```diff
@@ -0,0 +1,167 @@
+// rustfmt-indent_style: Visual
+
+fn long_parent() {
```

Comment:
This scenario was the most difficult to deal with (`Visual` for `indent_style` with the long element being the chain parent).

As I mentioned in the PR description this is definitely not perfect, but probably still better than not formatting the chains at all. I felt like this approach was the better of the evils, as doing the typical Visual indent with this scenario would basically result in the entire chain body being off screen (as all the child elements would be shifted to the right of the end of the long parent). 

In the future, it might be possible to leverage some of the heurstics added here to improve upon this scenario (for example checking whether the first child is long, etc.) but this feels like quite an edge case to me and I need a break from chains for a bit 😁